cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPATAuthenticationStrategy.java

```diff
@@ -30,15 +34,53 @@ public class GitPATAuthenticationStrategy implements AuthenticationStrategy {
   private static final String GITHUB_PAT_USERNAME = "oauth2";
 
   @Override
-  public CredentialsProvider getCredentialsProvider(SecureStore store,
-                                                    RepositoryConfig config,
-                                                    String namespaceId) throws AuthenticationConfigException {
-    try {
-      byte[] bytes = store.get(namespaceId, config.getAuth().getTokenName()).get();
-      String token = new String(bytes, StandardCharsets.UTF_8);
-      return new UsernamePasswordCredentialsProvider(GITHUB_PAT_USERNAME, token);
-    } catch (Exception e) {
-      throw new AuthenticationConfigException("Failed to get auth token from secure store", e);
+  public CredentialsProvider getCredentialsProvider(SecureStore store, RepositoryConfig config, String namespaceId) {
+    return new SecureStoreUserNamePasswordCredentialsProvider(store,
+                                                              GITHUB_PAT_USERNAME,
+                                                              config.getAuth().getTokenName(),
+                                                              namespaceId);
+  }
+
+  /**
+   * A class that wraps a {@link UsernamePasswordCredentialsProvider} but fetches password from a {@link SecureStore}.
+   */
+  private static class SecureStoreUserNamePasswordCredentialsProvider extends CredentialsProvider {
+    private final SecureStore secureStore;
+    private final String username;
+    private final String passwordTokenName;
+    private final String namespaceID;
+
+    SecureStoreUserNamePasswordCredentialsProvider(SecureStore secureStore,
+                                                   String username,
+                                                   String passwordTokenName,
+                                                   String namespaceID) {
+      this.secureStore = secureStore;
+      this.username = username;
+      this.passwordTokenName = passwordTokenName;
+      this.namespaceID = namespaceID;
+    }
+
+    @Override
+    public boolean isInteractive() {
+      return false;
+    }
+
+    @Override
+    public boolean supports(CredentialItem... credentialItems) {
+      return new UsernamePasswordCredentialsProvider("", "").supports(credentialItems);
+    }
+
+    @Override
+    public boolean get(URIish urIish, CredentialItem... credentialItems) throws UnsupportedCredentialItem {
+      try {
+        byte[] bytes = secureStore.get(namespaceID, passwordTokenName).get();
+        String token = new String(bytes, StandardCharsets.UTF_8);
+        return new UsernamePasswordCredentialsProvider(username, token).get(urIish, credentialItems);
+      } catch (IOException e) {
+        throw new IllegalStateException("Failed to get secret from secure store.", e);
+      } catch (Exception e) {
```

Comment:
Created https://cdap.atlassian.net/browse/CDAP-20365 for SecureStore interface improvements. The catch block was removed because of the refresh() method.